tests/utils/test_modeling_utils.py

```diff
@@ -1599,14 +1599,17 @@ def test_safetensors_torch_from_torch_sharded(self):
         for p1, p2 in zip(model.parameters(), new_model.parameters()):
             self.assertTrue(torch.equal(p1, p2))
 
-    def test_modifying_model_config_causes_warning_saving_generation_config(self):
+    def test_modifying_model_config_gets_moved_to_generation_config(self):
+        """
+        Calling `model.save_pretrained` should move the changes made to `generate` parameterization in the model config
+        to the generation config.
+        """
         model = AutoModelForCausalLM.from_pretrained("openai-community/gpt2")
-        model.config.top_k = 1
+        model.config.repetition_penalty = 3.0
         with tempfile.TemporaryDirectory() as tmp_dir:
-            with self.assertLogs("transformers.modeling_utils", level="WARNING") as logs:
-                model.save_pretrained(tmp_dir)
-            self.assertEqual(len(logs.output), 1)
-            self.assertIn("Your generation config was originally created from the model config", logs.output[0])
+            model.save_pretrained(tmp_dir)
+            self.assertTrue(model.config.repetition_penalty != 3.0)
```

Comment:
@amyeroberts that is a great question -- what gets written in `model.config` in this case. I agree with your comment (`If it's its old value, I'm worried this could cause confusion`), which was the actual behavior in the commit you reviewed, so I've made the following adjustments:
1. Moving a parameter from `model.config` to `model.generation_config` sets that parameter to `None` in `model.config`.
2. Storing a generation parameter as `None` in `model.config` is now accepted (even if it is not the default value);
3. Updated the warning message, explaining to the user why they are seeing it
4. The test now also checks that the warning is thrown. It is also commented, so it should be simple to understand what should be happening.
🤗